lib/src/container/mod.rs

```diff
@@ -246,6 +247,64 @@ impl std::fmt::Display for OstreeImageReference {
     }
 }
 
+/// Represent the difference in content between two OCI compliant Images
+#[derive(Debug, Default)]
+pub struct ManifestDiff {
+    all_layers_in_new: Vec<oci_spec::image::Descriptor>,
+    removed: Vec<oci_spec::image::Descriptor>,
+    added: Vec<oci_spec::image::Descriptor>,
+}
+
+/// Computes the difference between two OCI compliant images
+pub fn manifest_diff(
+    src: &oci_spec::image::ImageManifest,
+    dest: &oci_spec::image::ImageManifest,
+) -> ManifestDiff {
+    let src_layers = src
+        .layers()
+        .iter()
+        .map(|l| (l.digest(), l))
+        .collect::<HashMap<_, _>>();
+    let dest_layers = dest
+        .layers()
+        .iter()
+        .map(|l| (l.digest(), l))
+        .collect::<HashMap<_, _>>();
+    let mut diff = ManifestDiff::default();
+    for (blobid, &descriptor) in src_layers.iter() {
+        if !dest_layers.contains_key(blobid) {
+            diff.removed.push(descriptor.clone());
+        }
+    }
+    for (blobid, &descriptor) in dest_layers.iter() {
+        diff.all_layers_in_new.push(descriptor.clone());
+        if !src_layers.contains_key(blobid) {
+            diff.added.push(descriptor.clone());
+        }
+    }
+    diff
+}
+
+impl ManifestDiff {
+    /// Prints the total, removed and added content between two OCI images
+    pub fn print(&self) {
```

Comment:
I go back and forth on whether we should have code that calls `println!` in the library.  In some ways it feels architecturally unclean, but in practice it's...practical :smile:

Reply:
agreed